libp2p/protocols/rendezvous.nim

```diff
@@ -662,14 +684,36 @@ proc setup*(rdv: RendezVous, switch: Switch) =
   rdv.switch.addPeerEventHandler(handlePeer, Joined)
   rdv.switch.addPeerEventHandler(handlePeer, Left)
 
-proc new*(T: typedesc[RendezVous], rng: ref HmacDrbgContext = newRng()): T =
+proc new*(
+    T: typedesc[RendezVous],
+    rng: ref HmacDrbgContext = newRng(),
+    minDuration = MinimumDuration,
+    maxDuration = MaximumDuration,
+): T =
+  if minDuration < 1.minutes:
+    raiseAssert("TTL too short: 1 minute minimum")
+
+  if maxDuration > 72.hours:
+    raiseAssert("TTL too long: 72 hours maximum")
+
+  if minDuration >= maxDuration:
+    raiseAssert("Minimum TTL longer than maximum")
+
+  let
+    minTTL = minDuration.seconds.uint64
+    maxTTL = maxDuration.seconds.uint64
+
   let rdv = T(
     rng: rng,
     salt: string.fromBytes(generateBytes(rng[], 8)),
     registered: initOffsettedSeq[RegisteredData](1),
     defaultDT: Moment.now() - 1.days,
     #registerEvent: newAsyncEvent(),
```

Comment:
@lchenut just realized this. Do you remember why is this commented?

Reply:
I don't. I guess it's an artifact of an old implementation. Because it's not really useful afaik